R/serializer.R

```diff
@@ -497,26 +501,37 @@ self_set_serializer <- function(self, serializer) {
 #' The graphics device `dev_on` function will receive any arguments supplied to the serializer in addition to `filename`.
 #' `filename` points to the temporary file name that should be used when saving content.
 #' @param dev_off Function to turn off the graphics device. Defaults to [grDevices::dev.off()]
+#' @param serializer_params More docs needed here
+#'
 #' @export
-serializer_device <- function(type, dev_on, dev_off = grDevices::dev.off) {
+serializer_device <- function(type, dev_on, dev_off = grDevices::dev.off, serializer_params = list()) {
 
   stopifnot(!missing(type))
 
   stopifnot(!missing(dev_on))
   stopifnot(is.function(dev_on))
   stopifnot(length(formals(dev_on)) > 0)
   if (!any(c("filename", "...") %in% names(formals(dev_on)))) {
-    stop("`dev_on` must contain an arugment called `filename` or have `...`")
+    stop("`dev_on` must contain an argument called `filename` or have `...`")
   }
 
+  dev_requires_req <- "req" %in% names(formals(dev_on))
+
   stopifnot(is.function(dev_off))
 
   endpoint_serializer(
```

Comment:
Handling the `serializer_params`

** Untested

```suggestion
  serializer_params <- rlang::maybe_missing(serializer_params, {
      # Legacy support for static serializer parameters
      if (isTRUE(getOption("plumber.staticSerializers", default=FALSE)) return(list())
      # All known parameters for devices
      serializer_param_list()
    })

  endpoint_serializer(
```